activation/poet.go

```diff
@@ -338,15 +360,30 @@ func NewPoetClient(
 	logger *zap.Logger,
 	opts ...PoetClientOpt,
 ) (*poetClient, error) {
-	client, err := NewHTTPPoetClient(server, cfg, WithLogger(logger))
+	backend, err := NewHTTPPoetClient(server, cfg, WithLogger(logger))
 	if err != nil {
-		return nil, fmt.Errorf("creating HTTP poet client: %w", err)
-	}
+		return nil, fmt.Errorf("creating HTTP poet client %s: %w", server.Address, err)
+	}
+	return NewPoetClientWithBackend(
+		db,
+		backend,
+		cfg,
+		logger,
+		opts...,
+	), nil
+}
+
+func NewPoetClientWithBackend(
+	db poetDbAPI,
+	backend PoetBackend,
+	cfg PoetConfig,
+	logger *zap.Logger,
+	opts ...PoetClientOpt,
+) *poetClient {
```

Comment:
This doesn't need to be its own `New...` function. The `backend` can be passed into `NewPoetClient` as functional option which if not set defaults to what it is now.

Reply:
I prefer to have a separate constructor because creating an HTTP client requires additional parameters that are specific to it (`PoetServer`).